opencga-master/src/main/java/org/opencb/opencga/master/monitor/daemons/ExecutionDaemon.java

```diff
@@ -324,22 +332,46 @@ protected void checkJobs() throws CatalogException {
             }
             logger.info("----- EXECUTION DAEMON  ----- Organization={} --> pending={}, queued={}, running={}", organizationId, pendingJobs,
                     queuedJobs, runningJobs);
+            totalPendingJobs += pendingJobs;
+            totalQueuedJobs += queuedJobs;
+            totalRunningJobs += runningJobs;
         }
 
-        /*
-        PENDING JOBS
-        */
-        checkPendingJobs(organizationIds);
+        if (totalQueuedJobs == 0) {
+            // If there are no queued jobs, we can queue new jobs
+            List<Job> pendingJobs = new LinkedList<>();
+            List<Job> runningJobs = new LinkedList<>();
 
-        /*
-        QUEUED JOBS
-        */
-        checkQueuedJobs(organizationIds);
+            for (String organizationId : organizationIds) {
+                try (DBIterator<Job> iterator = jobManager.iteratorInOrganization(organizationId, pendingJobsQuery, queryOptions, token)) {
+                    while (iterator.hasNext()) {
+                        pendingJobs.add(iterator.next());
+                    }
+                } catch (Exception e) {
+                    logger.error("{}", e.getMessage(), e);
+                }
 
-        /*
-        RUNNING JOBS
-        */
-        checkRunningJobs(organizationIds);
+                try (DBIterator<Job> iterator = jobManager.iteratorInOrganization(organizationId, runningJobsQuery, queryOptions, token)) {
+                    while (iterator.hasNext()) {
+                        runningJobs.add(iterator.next());
+                    }
+                } catch (Exception e) {
+                    logger.error("{}", e.getMessage(), e);
+                }
+            }
+
+            jobScheduler.addJobs(pendingJobs, Collections.emptyList(), runningJobs);
```

Comment:
Pending and Running jobs are limited to NUM_JOBS_HANDLED=50, is this ok?

Reply:
I don't know... I just put that limit thinking we could have issues if someone sends thousands of jobs in one go (in theory, the only effect it should have is that it will be a bit slower but...). That's the reason why the options is also sorting by job priority... But we could start without setting any limits and see where that takes us.